string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,92 @@
+/* exported titleCase */
+
+// split arg title by ' ' into array
+// Enter loop over array
+// Switch statement for excluded words
+// exclude words with 4 characters
+// Process 'api' as 'API'
+// Process varieties of the word "JavaScript"
+// included spcial exceptions
+
+function titleCase(title) {
+  var titleSplit = title.split(' ');
+  var titleFinal = [];
+
+  for (var i = 0; i < titleSplit.length; i++) {
+    switch (titleSplit[i]) {
+
+      case 'and':
+      case 'or':
+      case 'nor':
+      case 'but':
+      case 'a':
+      case 'as':
+      case 'at':
+      case 'by':
+      case 'for':
+      case 'in':
+      case 'of':
+      case 'on':
+      case 'per':
+      case 'to' :
+        titleFinal.push(titleSplit[i]);
+        break;
+      case 'an':
+
+        if (i === 0) {
+          titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+        } else if (titleSplit[i - 1][titleSplit[i - 1].length - 1] === ':') {
+          titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+        } else {
+          titleFinal.push(titleSplit[i]);
+        }
+        break;
+
+      case 'the':
+        if (i === 0) {
+          titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+        } else if (titleSplit[i - 1][titleSplit[i - 1].length - 1] === ':') {
+          titleFinal.push(titleSplit[i][0].toUpperCase() + titleSplit[i].slice(1, titleSplit[i].length));
+        } else {
+          titleFinal.push(titleSplit[i]);
+        }
```

Comment:
Why are "an" and "the" doing the work for words with colons in them? This is not going to work across all situations, you are only accounting for colons in the case of the word "an" and "the". You should deal with colons in a separate space than where you are supposedly dealing with excluded words so it will work across all possible situations.